internal/pkg/aws/codepipeline/codepipeline_test.go

```diff
@@ -505,3 +505,180 @@ func TestCodePipeline_GetPipelineState(t *testing.T) {
 		})
 	}
 }
+
+func TestCodePipeline_ListPipelineExecution(t *testing.T) {
```

Comment:
nit: since now `listPipelineExecution` is private we can instead think about baking some of these tests inside the Retry test

Reply:
Oh yeah! Done.